workflow/packer/packer.go

```diff
@@ -59,7 +59,7 @@ func IsTooLargeError(err error) bool {
 	return err != nil && strings.HasPrefix(err.Error(), tooLarge)
 }
 
-func CompressWorkflow(wf *wfv1.Workflow) error {
+func CompressWorkflowIfNeeded(wf *wfv1.Workflow) error {
```

Comment:
Changed the name to better reflect that this function is a no-op if compression is not needed.